marshmallow/schema.py

```diff
@@ -766,6 +766,8 @@ def _invoke_field_validators(self, data, many):
             try:
                 field_obj = self.fields[field_name]
             except KeyError:
+                if field_name in self.declared_fields:
+                    return
```

Comment:
I think you want to `continue` here instead of `return`ing. Otherwise other declared validators may not be invoked.